File: leetcode-submissions/2270_Number_of_Ways_to_Split_Array.ts
function waysToSplitArray(nums: number[]): number {
    const prefixSum:number[]=Array.from({length:nums.length},()=>0);
    const suffixSum:number[]=Array.from({length:nums.length},()=>0);
    prefixSum[0]=nums[0];
    suffixSum[nums.length-1]=nums[nums.length-1];

    for(let i=1;i<nums.length;i++){
        prefixSum[i]=prefixSum[i-1]+nums[i];
    }
    for(let i=nums.length-2;i>=0;i--){
        suffixSum[i]=suffixSum[i+1]+nums[i];
    }
    let count =0;
    for(let i:number =0;i<prefixSum.length;i++){
        if(prefixSum[i]>=suffixSum[i+1]){
            count ++;
        }
    }
    return count ;

};

File: leetcode-submissions/2270_Number_of_Ways_to_Split_Array.cpp
class Solution {
public:
    int waysToSplitArray(vector<int>& nums) {
        vector<int>prefixSum(nums.size(),0);
        vector<int>suffixSum(nums.size(),0);
        prefixSum[0]=nums[0];
        for(int i=1;i<nums.size();i++){
            prefixSum[i]=prefixSum[i-1]+nums[i];
        }
        suffixSum[nums.size()-1]=nums[nums.size()-1];
        for(int i=nums.size()-2;i>=0;i--){
            suffixSum[i]=suffixSum[i+1]+nums[i];
        }
        int count=0;
        for(int i=0;i<prefixSum.size()-1;i++){
            if(prefixSum[i]>=suffixSum[i+1]){
                count++;
            }
        }
        return count ;
    }
};